### PR TITLE
Export national survey data

### PR DESCRIPTION
**Pull request by kevin-verschaeve**
[ch12440]

### CODE REVIEW

File: src/Repository/Jecoute/DataSurveyRepository.php [hunk 1]
```diff
@@ -30,4 +31,14 @@ public function countByEmailAnsweredForOneMonth(string $email, \DateTime $posted
             ->getSingleScalarResult()
         ;
     }
+
+    public function iterateForSurvey(Survey $survey)
```
Comment (ottaviano) [inline src/Repository/Jecoute/DataSurveyRepository.php:35]: typehint

File: templates/admin/jecoute/exports.html.twig [hunk 2]
```diff
@@ -0,0 +1,14 @@
+{% extends 'SonataAdminBundle:CRUD:base_list_field.html.twig' %}
+
+{% block field %}
+    {% if admin.hasAccess('show') %}
```
Comment (ottaviano) [inline templates/admin/jecoute/exports.html.twig:4]: do we need to do the same check in the controller?

File: src/Exporter/NationalSurveyExporter.php [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+                (new Slugify())->slugify($survey->getName()),
+                $survey->getId(),
+                (new \DateTime())->format('YmdHis'),
+                $format
+            ),
+            new IteratorCallbackSourceIterator($this->dataSurveyRepository->iterateForSurvey($survey), function (array $data) use ($survey, $allowedOnly) {
+                /** @var DataSurvey $dataSurvey */
+                $dataSurvey = $data[0];
+
+                $allowPersonalData = ($allowedOnly && $dataSurvey->getAgreedToTreatPersonalData()) || !$allowedOnly;
```
Comment (ottaviano) [inline src/Exporter/NationalSurveyExporter.php:49]: is it the same? 
```php
$allowPersonalData = !$allowedOnly || $dataSurvey->getAgreedToTreatPersonalData();
```

File: templates/admin/jecoute/exports.html.twig (hunk 2 above)
Comment (kevin-verschaeve) [reply templates/admin/jecoute/exports.html.twig:4]: yes indeed, and I actually moved this check into the Admin class, so the column is not added if we don't have the access

**Review by ottaviano — APPROVED** [review]

**Pull request merged**